### PR TITLE
C-1989 Fix "Download is not re-starting even though there are still many tracks that need to be downloaded"

### PR DESCRIPTION
**Pull request by nicoback2**
### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/services/offline-downloader/offline-sync.ts [hunk 1]
```diff
@@ -144,8 +144,18 @@ export const syncCollectionTracks = async (
   // TODO: record and check last verified time for collections
   const state = store.getState()
   const currentUserId = getUserId(state as unknown as CommonState)
+  const downloadedTracks = getOfflineTracks(state)
   const collectionId = offlineCollection.playlist_id
   const collectionIdStr = offlineCollection.playlist_id.toString()
+  const queuedTracks = (await queue.getJobs())
```
Comment (dylanjeffers) [inline packages/mobile/src/services/offline-downloader/offline-sync.ts:150]: queuedTrackIds

File: packages/mobile/src/services/offline-downloader/offline-sync.ts [hunk 2]
```diff
@@ -156,32 +166,20 @@ export const syncCollectionTracks = async (
...
-  )
-  if (updatedCollection.cover_art_sizes !== offlineCollection.cover_art_sizes) {
-    downloadCollectionCoverArt(updatedCollection)
-  }
-
-  const oldTrackIds = new Set(
-    offlineCollection.tracks?.map((track) => track.track_id)
-  )
+  const oldTrackIds = new Set([
+    ...(offlineCollection.tracks
```
Comment (dylanjeffers) [inline packages/mobile/src/services/offline-downloader/offline-sync.ts:170]: can we do the map filter separtely in a const, like offlineCollectionTrackIds?, then oldTrackIds is a bit easier to read? nit for sure

File: packages/mobile/src/services/offline-downloader/offline-sync.ts [hunk 3]
```diff
@@ -156,32 +166,20 @@ export const syncCollectionTracks = async (
...
+      ?.map((track) => track.track_id)
+      .filter((trackId) => downloadedTracks[trackId] != null) || []),
+    ...queuedTracks
+  ])
   const newTrackIds = new Set(
     updatedCollection.tracks?.map((track) => track.track_id)
   )
+
   const addedTrackIds = [...newTrackIds].filter(
```
Comment (dylanjeffers) [inline packages/mobile/src/services/offline-downloader/offline-sync.ts:179]: dont need a spread here cause filter creates new array!

**Review by dylanjeffers — APPROVED** [review]
clutch! verified this works too. just some minor things around formatting/readability

File: packages/mobile/src/services/offline-downloader/offline-sync.ts (hunk 3 above)
Comment (nicoback2) [reply packages/mobile/src/services/offline-downloader/offline-sync.ts:178]: need it because its a set :P

Comment (dylanjeffers) [reply packages/mobile/src/services/offline-downloader/offline-sync.ts]: Oh yeah!

**Pull request merged**